src/ol/geom/point.js

```diff
@@ -104,7 +104,7 @@ ol.geom.Point.prototype.intersectsExtent = function(extent) {
  */
 ol.geom.Point.prototype.setCoordinates = function(coordinates, opt_layout) {
   if (!coordinates) {
-    this.setFlatCoordinates(ol.geom.GeometryLayout.XY, null);
+    this.setFlatCoordinates(opt_layout || ol.geom.GeometryLayout.XY, null);
```

Comment:
I think it's more correct to respect the passed parameter, if present.
It is also needed to keep dimension information for empty (i.e. POINT ZM EMPTY).
Se failing tests if you remove it.